Horizontal bars for cpu/mem resources of pods are no longer displayed in some views since 4.2.3

**Describe the bug**
Horizontal bars for cpu/mem resources of pods are no longer displayed in some views since 4.2.3.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to deployment view.
2. Horizontal bars are not displayed for pods.

**Expected behavior**
Horizontal bars showing the CPU/mem usage should be visible.

**Screenshots**
Deployment view:
![image](https://user-images.githubusercontent.com/52334444/116457310-3d492600-a831-11eb-98ed-0b47ca9fd836.png)

Node view:
![image](https://user-images.githubusercontent.com/52334444/116457826-e263fe80-a831-11eb-9a33-4a5d1909686e.png)

**Environment (please complete the following information):**
- Lens Version: 4.2.3
- OS: macOS
- Installation method (e.g. snap or AppImage in Linux): artifact from GH release long time ago then auto-update since then.

**Logs:**
Logs from the dev tool console:
```
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/lens-cluster-store.json
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/lens-extensions.json
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/lens-filesystem-provisioner-store.json
console.js:124 info: [CLUSTER-STORE] requesting initial state sync
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/lens-workspace-store.json
console.js:124 info: [WORKSPACE-STORE] requesting initial state sync
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/lens-user-store.json
console.js:124 info: [CLUSTER-STORE] start to listen (1)
console.js:124 info: [WORKSPACE-STORE] starting to listen state events
console.js:124 info: [EXTENSION]: enabled kube-object-event-status@0.1.0
console.js:124 info: [EXTENSION]: enabled lens-metrics-cluster-feature@0.1.0
console.js:124 info: [EXTENSION]: enabled lens-node-menu@0.1.0
console.js:124 info: [EXTENSION]: enabled lens-pod-menu@0.1.0
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/extension-store/lens-survey/preferences-store.json
console.js:124 info: [EXTENSION]: enabled lens-survey@0.1.0
/Applications/Lens.app/Contents/Resources/extensions/telemetry/dist/renderer.js:121 telemetry extension activated
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/extension-store/lens-telemetry/preferences-store.json
console.js:124 info: [EXTENSION]: enabled lens-telemetry@0.1.0
console.js:124 info: [PROTOCOL ROUTER]: internal registering /preferences
console.js:124 info: [PROTOCOL ROUTER]: internal registering /
console.js:124 info: [PROTOCOL ROUTER]: internal registering /landing
console.js:124 info: [PROTOCOL ROUTER]: internal registering /landing/:workspaceId
console.js:124 info: [PROTOCOL ROUTER]: internal registering /cluster
console.js:124 info: [PROTOCOL ROUTER]: internal registering /cluster/:clusterId
console.js:124 info: [PROTOCOL ROUTER]: internal registering /cluster/:clusterId/settings
console.js:124 info: [PROTOCOL ROUTER]: internal registering /extensions
console.js:124 info: [PROTOCOL ROUTER]: internal registering /extensions/install/:LENS_INTERNAL_EXTENSION_PUBLISHER_MATCH(@[A-Za-z0-9_]+)?/:LENS_INTERNAL_EXTENSION_NAME_MATCH
console.js:124 info: [LENS-VIEW]: init dashboard, clusterId=7b803157-ca51-437e-bd99-bf68c9a33a82
console.js:124 info: [LENS-VIEW]: waiting cluster to be ready, clusterId=7b803157-ca51-437e-bd99-bf68c9a33a82
console.js:124 error: Can't find working API for the Kubernetes resource ingresses
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/lens-cluster-store.json
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/lens-extensions.json
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/lens-filesystem-provisioner-store.json
console.js:124 info: [CLUSTER-STORE] requesting initial state sync
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/lens-workspace-store.json
console.js:124 info: [WORKSPACE-STORE] requesting initial state sync
console.js:124 info: [LENS-VIEW]: loaded from http://7b803157-ca51-437e-bd99-bf68c9a33a82.localhost:52566/
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/lens-user-store.json
console.js:124 info: [CLUSTER-STORE] start to listen (5)
console.js:124 info: [WORKSPACE-STORE] starting to listen state events
console.js:124 info: [APP]: Init dashboard, clusterId=7b803157-ca51-437e-bd99-bf68c9a33a82, frameId=5
mobx.module.js:3369 [mobx.array] Attempt to read an array index (0) that is out of bounds (0). Please check length first. Out of bound indices will not be tracked by MobX
get @ Lens.js:2
console.js:124 info: [EXTENSION]: enabled kube-object-event-status@0.1.0
console.js:124 info: [EXTENSION]: enabled lens-metrics-cluster-feature@0.1.0
console.js:124 info: [EXTENSION]: enabled lens-node-menu@0.1.0
console.js:124 info: [EXTENSION]: enabled lens-pod-menu@0.1.0
console.js:124 info: [EXTENSION]: enabled lens-survey@0.1.0
/Applications/Lens.app/Contents/Resources/extensions/telemetry/dist/renderer.js:121 telemetry extension activated
console.js:124 info: [STORE]: LOADED from /Users/maxenceboutet/Library/Application Support/Lens/extension-store/lens-telemetry/preferences-store.json
console.js:124 info: [EXTENSION]: enabled lens-telemetry@0.1.0
pods.store.ts:31 loadKubeMetrics failed TypeError: Cannot read property 'length' of null
    at e.spliceWithArray (mobx.module.js:3187)
    at Proxy.replace (mobx.module.js:3271)
    at bm.loadKubeMetrics (pods.store.ts:29)
loadKubeMetrics @ Lens.js:2
async function (async)
loadKubeMetrics @ Lens.js:2
(anonymous) @ Lens.js:2
s @ Lens.js:2
start @ Lens.js:2
componentDidMount @ Lens.js:2
uo @ Lens.js:2
Qc @ Lens.js:2
t.unstable_runWithPriority @ Lens.js:2
zn @ Lens.js:2
Ic @ Lens.js:2
hc @ Lens.js:2
(anonymous) @ Lens.js:2
t.unstable_runWithPriority @ Lens.js:2
zn @ Lens.js:2
Hn @ Lens.js:2
Gn @ Lens.js:2
Fe @ Lens.js:2
Kt @ Lens.js:2
pods.store.ts:31 loadKubeMetrics failed TypeError: Cannot read property 'length' of null
    at e.spliceWithArray (mobx.module.js:3187)
    at Proxy.replace (mobx.module.js:3271)
    at bm.loadKubeMetrics (pods.store.ts:29)
loadKubeMetrics @ Lens.js:2
async function (async)
loadKubeMetrics @ Lens.js:2
(anonymous) @ Lens.js:2
s @ Lens.js:2
setInterval (async)
start @ Lens.js:2
componentDidMount @ Lens.js:2
uo @ Lens.js:2
Qc @ Lens.js:2
t.unstable_runWithPriority @ Lens.js:2
zn @ Lens.js:2
Ic @ Lens.js:2
hc @ Lens.js:2
(anonymous) @ Lens.js:2
t.unstable_runWithPriority @ Lens.js:2
zn @ Lens.js:2
Hn @ Lens.js:2
Gn @ Lens.js:2
Fe @ Lens.js:2
Kt @ Lens.js:2
```

**Kubeconfig:**
Not relevant and has not changed in months.

**Additional context**
Using `prometheus-operator` from the Bitnami `kube-prometheus` chart installed in the `prometheus` namespace. Set to "auto-detect" in Lens.

--- a/src/renderer/api/__tests__/kube-object.test.ts
+++ b/src/renderer/api/__tests__/kube-object.test.ts
@@ -110,19 +110,6 @@ describe("KubeObject", () => {
     {
       type TestCase = [string, any];
       const tests: TestCase[] = [
-        ["metadata.uid", { kind: "", apiVersion: "", metadata: {  name: "", resourceVersion: "", selfLink: ""} }],
-        ["metadata.name", { kind: "", apiVersion: "", metadata: { uid: "", resourceVersion: "", selfLink: "" } }],
-        ["metadata.resourceVersion", { kind: "", apiVersion: "", metadata: { uid: "", name: "", selfLink: "" } }],
-      ];
-
-      it.each(tests)("should reject with missing non-top level field: %s", (missingField, input) => {
-        expect(KubeObject.isPartialJsonApiData(input)).toBe(false);
-      });
-    }
-
-    {
-      type TestCase = [string, any];
-      const tests: TestCase[] = [
         ["kind", { kind: 1, apiVersion: "", metadata: { uid: "", name: "", resourceVersion: "", selfLink: "" } }],
         ["apiVersion", { apiVersion: 1, kind: "", metadata: { uid: "", name: "", resourceVersion: "", selfLink: "" } }],
         ["metadata", { kind: "", apiVersion: "", metadata: "" }],
@@ -192,7 +179,6 @@ describe("KubeObject", () => {
         ["kind", { apiVersion: "", items: [], metadata: { resourceVersion: "", selfLink: "" } }],
         ["apiVersion", { kind: "", items: [], metadata: { resourceVersion: "", selfLink: "" } }],
         ["metadata", { kind: "", items: [], apiVersion: "" }],
-        ["metadata.resourceVersion", { kind: "", items: [], apiVersion: "", metadata: { selfLink: "" } }],
       ];
 
       it.each(tests)("should reject with missing: %s", (missingField, input) => {

--- a/src/renderer/api/endpoints/pod-metrics.api.ts
+++ b/src/renderer/api/endpoints/pod-metrics.api.ts
@@ -2,7 +2,7 @@ import { KubeObject } from "../kube-object";
 import { KubeApi } from "../kube-api";
 
 export class PodMetrics extends KubeObject {
-  static kind = "Pod";
+  static kind = "PodMetricsList";
   static namespaced = true;
   static apiBase = "/apis/metrics.k8s.io/v1beta1/pods";
 

--- a/src/renderer/api/endpoints/pod-metrics.api.ts
+++ b/src/renderer/api/endpoints/pod-metrics.api.ts
@@ -2,7 +2,7 @@ import { KubeObject } from "../kube-object";
 import { KubeApi } from "../kube-api";
 
 export class PodMetrics extends KubeObject {
-  static kind = "PodMetricsList";
+  static kind = "PodMetrics";
   static namespaced = true;
   static apiBase = "/apis/metrics.k8s.io/v1beta1/pods";
 

--- a/src/renderer/api/kube-object.ts
+++ b/src/renderer/api/kube-object.ts
@@ -91,7 +91,7 @@ export class KubeObject implements ItemObject {
   static isKubeJsonApiListMetadata(object: unknown): object is KubeJsonApiListMetadata {
     return (
       isObject(object)
-      && hasTypedProperty(object, "resourceVersion", isString)
+      && hasOptionalProperty(object, "resourceVersion", isString)
       && hasOptionalProperty(object, "selfLink", isString)
     );
   }
@@ -112,12 +112,28 @@ export class KubeObject implements ItemObject {
     );
   }
 
+  static isPartialJsonApiMetadata(object: unknown): object is Partial<KubeJsonApiMetadata> {
+    return (
+      isObject(object)
+      && hasOptionalProperty(object, "uid", isString)
+      && hasOptionalProperty(object, "name", isString)
+      && hasOptionalProperty(object, "resourceVersion", isString)
+      && hasOptionalProperty(object, "selfLink", isString)
+      && hasOptionalProperty(object, "namespace", isString)
+      && hasOptionalProperty(object, "creationTimestamp", isString)
+      && hasOptionalProperty(object, "continue", isString)
+      && hasOptionalProperty(object, "finalizers", bindPredicate(isTypedArray, isString))
+      && hasOptionalProperty(object, "labels", bindPredicate(isRecord, isString, isString))
+      && hasOptionalProperty(object, "annotations", bindPredicate(isRecord, isString, isString))
+    );
+  }
+
   static isPartialJsonApiData(object: unknown): object is Partial<KubeJsonApiData> {
     return (
       isObject(object)
       && hasOptionalProperty(object, "kind", isString)
       && hasOptionalProperty(object, "apiVersion", isString)
-      && hasOptionalProperty(object, "metadata", KubeObject.isKubeJsonApiMetadata)
+      && hasOptionalProperty(object, "metadata", KubeObject.isPartialJsonApiMetadata)
     );
   }
 

